internal/provider/kubernetes/gateway.go

```diff
@@ -377,7 +377,12 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
 	if !found {
 		gw, ok := r.resources.Gateways.Load(request.NamespacedName)
 		if !ok {
-			r.log.Info("failed to find gateway in the watchable map", "namespace", gw.Namespace, "name", gw.Name)
+			if gw != nil {
```

Comment:
not needed, since its already `!ok`

Reply:
So I think we should return directly here?

Reply:
When we can load gw from watchable map, we should not let logic to go down, which will cause more panics, I think.